### ISSUE TITLE
Store entry event study

### ISSUE THREAD

**Issue by huntallcott** [huntallcott/FoodDeserts#9]
The goal of this issue is to carry out and discuss the revised store entry event study.
@handbury and I agreed earlier this week that at least to start, we will replicate ADD's Figure 5 with TDLinx data.
@handbury you have posted the TDLinx store list on dropbox. The easy thing to do is just limit to the sample with stdlinxyear=2016 and use the Date_Added as the entry date for all stores. But didn't you have some code that allowed you to screen out/drop rows that were relocations or replacements instead of actual new openings?  For Walmart this is definitely relevant. And do you have some other advice/code here?
Thanks.

**Comment by handbury** [comment]
@huntallcott I have pulled out the code that we used to identify potential M&A activity/relocations/replacements.  It's in the attached code.

The code reads in two datasets:
1. USDAJune2015Changes.txt: from the USDA listing stores whose characteristics changed in the TDLinx sample (indicating, say, a merger or acquisition).
2. tdlinx_blockids.dta: a datasaet where we filled in the block id variable based on the lat-long coordinates in the TDLinx data.
3. The store open datasets from which I created the csv store list that you have already.

1 through 3 are in Dropbox: Externals\Data\StoreEntryExitAll three types of data. 

I commited the code into GitHub\FoodDeserts\Code\DataPrep\Supply.  Note that it limits attention to the entries between 2006 and 2011, inclusive, though we have data for entries up to 2016.  It should run fine if change the paths up top.

Let me know if you have any questions.

Thanks!

**Comment by huntallcott** [comment]
@handbury quick follow-up. I see the files on dropbox but don't see any new code in GitHub's Code/DataPrep/Supply folder. I looked in the [history](https://github.com/huntallcott/FoodDeserts/commits/master/Code/DataPrep/Supply), and I don't think you pushed anything to the master ever.
Can you double check that you pushed to the master? You have to both "commit" (the blue bar at bottom left of GitHub desktop) and then press the "Push origin" button near top right to push that commit to the master. 
Many thanks.

**Comment by handbury** [comment]
@huntallcott Sorry about that.  I've pushed the commit through.

**Comment by huntallcott** [comment]
@handbury the file breaks on what is now line 294:
`merge m:1 sblockid prevyear using `tempblock', keep(1 3)`
I checked and I'm not sure that tempblock is created in the same script. Could you take a look? I'm not editing the file, so you can edit anywhere in the text.

**Comment by huntallcott** [comment]
Also @handbury in what is now line 327 there is a local called `dataset' that I think will not work. Can you look at that as well?
Many thanks and sorry for my confusion in making this work.

**Comment by handbury** [comment]
@huntallcott (@rakhkovs and @mollyschnell): this stdlinxscd-store_code_uc concordance is derived from a USDA version of the Homescan data that contains the stdlinxscd for each store in the data (now in Externals\Data\Nielsen\Homescan\td_hm.dta) .  I expect that you can use this data to create a cleaner store_code_uc-stdlinxscd concordance than we were able to, since we only observed store 5-digit zip codes. @rakhkovs can provide more details on how TDLinx codes were assigned to stores (identified by StoreName in the td_hm dataset). 

As the TDLinx changes dataset I uploaded to the store entry directory last week suggests, the characteristics can change over time for a given stdlinxscd.  Presumably, this means that a new store_code_uc might be assigned to a store without the stdlinxscd changing (after a merger or acquisition results in a name and/or ownership change, for example).  This might be the case in some of the many-to-one matches.

**Comment by huntallcott** [comment]
@handbury , in Externals\Data\Nielsen\Homescan\td_hm.dta, I think that storename is the retailer_code, not the store_code_uc. 
Let me email JP and Rebecca to see if they have another concordance, possibly one that goes through 2016.

**Comment by handbury** [comment]
@huntallcott - yes, StoreName is the retailer, not store_code_uc (which is a Kilts-created variable, so not in the USDA version of the Homescan data).  We collapse td_hm.dta to the purchase date, household, store name (retailer) level and merge it into the Kilts HMS data to get the store_code_uc, dropping cases where we see a household shopping in two outlets of the same retailer on the same day.

**Comment by rakhkovs** [comment]
@huntallcott For early to mid-2000s Nielsen matched its stores with TDLinx. In the Homescan form consumers only write the name of the store. If a consumer shopped in Safeway, Nielsen does not know which Safeway it was. Nielsen matched the sale with the nearest Safeway to the consumer's home from TDLinx. For the late 2000s when Nielsen stopped providing us the TDLinx codes we followed the same procedure ourselves.

**Comment by huntallcott** [comment]
I found a tdlinx-store_code_uc concordance from Rebecca or JP that matches about 90% of the entries.

**Comment by handbury** [comment]
Great!

**Comment by huntallcott** [comment]
@handbury you noted above that the file limits attention to 2006-2011. I just changed line 288 of the current code to read:
keep if entry_year>=2004 & entry_year<=$MaxYear

The file runs fine, but I just wanted to confirm: is that OK with you or will this generate some other problem that I might not foresee, in terms of the data cleaning or something else?

**Comment by huntallcott** [comment]
@handbury with apologies I don't fully understand that. Would you be willing to pull down the most recent code from the master and check whether the block IDs are "filled in"? Would be a big help.
One thing I did change is line 263, now reading keep if year >= 2004 & year <= $MaxYear.

**Comment by handbury** [comment]
@huntallcott No problem.  The merge is pretty complete.  We identify blockids for 1245656 out of 1273043 TDLinx store listings.  FYI: the sblockidraw is the block id that was provided in the raw TDLinx data; sblockid is the 2010 block id that we pulled based on the lat/long coordinates provided.

**Comment by huntallcott** [comment]
@handbury @mollyschnell @diamondr ,
First, thanks to @diamondr for re-doing drive times, to @handbury for working on the TDLinx entry list, and to @moosedube for getting the appropriate store_code_ucs. 
I am now re-running the store entry event study with the data. I need a few more hours for some things, but you can see the updated version of the main figure [here](https://github.com/huntallcott/FoodDeserts/blob/master/Output/ReducedForm/Figures/Entry1Figures40810.pdf).
Same qualitative results: people shop at entrant stores, but this mostly cannibalizes expenditures from other grocery stores, so the food environment doesn't change, and then in the bottom figures you see that the Health Index doesn't change. 
The figures look way nicer, and in particular the standard errors are half as wide, because the TDLinx data have many more entries. 
One thing to note is that the entrant expenditures only go up by 2.5 percentage points (top left panel), whereas that number was 6 percentage points before. I think that this is because we previously were looking primarily at supercenters, which are huge, and now we are looking at a wider sample of grocery stores. (Our entrant sample doesn't include convenience stores or drug stores, but it does include grocery stores, supercenters, and club stores -- everything that ADD had called "supermarkets" -- and many of these grocery stores are smaller than supercenters.) It could also be due to measurement error in store entries, although I'm pretty confident that this is not the case, or due to measurement error in the trips data store codes, which does seem quite plausible.
I may consider two things:
1. Redo the entrant expenditures using expenditures for the entrant retailer_code. This could address measurement error in trips data store codes.
2. Now that we have more precision, focus the entire figure on entry into food deserts.

Post here if any thoughts?

**Comment by handbury** [comment]
@huntallcott It's possible that measurement error in the trips store codes is biasing the entrant expenditures downwards.  Since Nielsen attributes the nearest store with the same retailer code to any given trip, it probably makes sense to switch it to looking at "entrant chain" expenditures to check.

I also think that it's worthwhile looking at food deserts independently.  If grocery/super/club expenditures only increase marginally, perhaps that plot can be replaced with the "entrant expenditure" plot for the food desert sub-sample to mirror the lower panel.

Thanks for re-running all of this!

**Comment by huntallcott** [comment]
@handbury , great, thanks, and am running these now. These take a bit of time but I will have an update over the weekend.

**Comment by huntallcott** [comment]
@mollyschnell @handbury @diamondr @rakhkovs @moosedube ,
Just a quick update on the entry event study. I have been working over the past few days on improving the retailer name matching between TDLinx and Nielsen. I think that this is figured out and I should have some revised results over the next 48 hours.

**Comment by huntallcott** [comment]
Hi all,
I have just finished revising the stylized facts, movers, and entry event study. The figures and tables in the lyx and pdf files, including table/figure notes, are updated. Take a look? If there are no objections, then I will finish revising the text.

**Comment by huntallcott** [comment]
@mollyschnell @handbury @Rahkovsky @diamondr @moosedube ,
I have finished updating the entry analysis with the HEI. 
* Here the analysis is broadly similar in the big picture, although we see some statistically detectable effects of entry on Health Index in food deserts, in various alternative definitions of what a food desert is. (We had one significant result before, but now there are more.) So I changed the text around that. 
* The big picture is still the same. We have TREMENDOUS power in this paper, and more than three times more supermarket entries than those suckers ADD had before, and so we are able to pick up even economically insignificant effects.
